tpls/runtime-template.js

```diff
@@ -204,5 +204,22 @@ function applyUpdate(callback, errback) {
   <% } %>
 }
 
+function updateOfflineService() {
+  if (navigator.serviceWorker) {
+    navigator.serviceWorker.ready.then((registration) => {
+      if(!registration) {
+        return;
+      }
+      return registration.update();
+    });
+  } else if (window.applicationCache) {
+    applicationCache.update();
```

Comment:
I don't remember why, but I call application cache methods on iframe instance rather than on top instance. There probably should be a reason for that, so would be good to have it that way.